repo_remote.go

```diff
@@ -146,3 +146,181 @@ func RepoRemoveRemote(repoPath, name string, opts ...RemoveRemoteOptions) error
 func (r *Repository) RemoveRemote(name string, opts ...RemoveRemoteOptions) error {
 	return RepoRemoveRemote(r.path, name, opts...)
 }
+
+// RemotesListOptions contains arguments for listing remotes of the repository.
+// Docs: https://git-scm.com/docs/git-remote#_commands
+type RemotesListOptions struct {
+	// The timeout duration before giving up for each shell command execution.
+	// The default timeout duration will be used when not supplied.
+	Timeout time.Duration
+}
+
+// RepoRemotesList lists remotes of the repository in given path.
+func RepoRemotesList(repoPath string, opts ...RemotesListOptions) ([]string, error) {
+	var opt RemotesListOptions
+	if len(opts) > 0 {
+		opt = opts[0]
+	}
+
+	stdout, err := NewCommand("remote").RunInDirWithTimeout(opt.Timeout, repoPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return stdoutToStringSlice(stdout), nil
+}
+
+// RemotesList lists remotes of the repository.
+func (r *Repository) RemotesList(opts ...RemotesListOptions) ([]string, error) {
+	return RepoRemotesList(r.path, opts...)
+}
+
+// RemoteURLGetOptions contains arguments for retrieving URL(s) of a remote of the repository.
+// Docs: https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emget-urlem
+type RemoteURLGetOptions struct {
+	// False: get fetch URLs
+	// True: get push URLs
+	Push bool
```

Comment:
Godoc converts the line breaks into spaces, so this is how it looks like in hover-over:
<img width="281" alt="CleanShot 2022-02-10 at 12 44 53@2x" src="https://user-images.githubusercontent.com/2946214/153338769-85077f4c-4ceb-4470-baa3-a045512c61c7.png">

```suggestion
	// Indicates whether to get push URLs instead of fetch URLs.
	Push bool
```